stories/src/API/repository/story.repository.ts

```diff
@@ -1,31 +1,32 @@
+import { CreateStoryDTO } from "../../dtos/create-story.dto.js";
+// Story entity is used when all properties for a created story on the database are mandatory
+import { StoryEntity } from "../../entities/story.entity.js";
+// same structure as the Story Entity but it has some optional database properties 
+import { StoryInterface } from "../../interfaces/interfaces.js";
+
 export default class StoryRepository {
-  constructor(private _database: any) {}
+  constructor(private readonly database: any) {}
 
-  async create(story: Object) {
-    return await this._database.Stories.create(story);
+  async create(storyData: CreateStoryDTO): Promise<StoryEntity> {
+    return await this.database.Stories.create(storyData);
   }
 
-  async update(storyId: string, data: any) {
-    const updatedStory = await this._database.Stories.update(...data, {
+  async update(storyId: string, data: Partial<StoryInterface>): Promise<StoryEntity> {
+    const updatedStory = await this.database.Stories.update(data, {
```

Comment:
Em quase todos os métodos tem `this.databasae.Stories.OPERATION`, Acho que vale a pena guardar uma referência direto para as stories para as chamadas serem encurtadas para `this.stories.OPERATION`